plugins/inputs/prometheus/prometheus.go

```diff
@@ -200,7 +200,15 @@ func (p *Prometheus) GetAllURLs() (map[string]URLAndAddress, error) {
 			return allURLs, err
 		}
 
-		for _, url := range getMesosTaskPrometheusURLs(tasks) {
+		u, uerr := url.Parse(p.MesosAgentUrl)
+		hostname, _, serr := net.SplitHostPort(u.Host)
+		if uerr != nil || serr != nil {
+			log.Printf("E! could not retrieve hostname from mesos agent url: %s %s", uerr, serr)
+			log.Printf("W! mesos task prometheus metrics will be pulled from 'localhost'")
+			hostname = "localhost"
```

Comment:
If either of these errors occurs then it means there's an invalid value in Telegraf config. It'd be easier to debug that if Telegraf died right after it started with a clear error message. How about we parse the `MesosAgentUrl` in `Start()`, and exit with an error if it can't be parsed?